### PR TITLE
chore: upgrade http spec and remove hack

### PR DESCRIPTION
**Pull request by XVincentX**

### CODE REVIEW

File: examples/petstore.oas2.yaml [hunk 1]
```diff
@@ -1,700 +1,19 @@
 swagger: '2.0'
-info:
```
Comment (lag-of-death) [inline examples/petstore.oas2.yaml]: @XVincentX, why are we removing this?

File: test-harness/specs/validate-body-params/no-consumes.defined.oas2.txt [hunk 2]
```diff
@@ -1,37 +0,0 @@
-====test====
-When I send a request to an operation that does not have a consumes defined
```
Comment (lag-of-death) [inline test-harness/specs/validate-body-params/no-consumes.defined.oas2.txt]: @XVincentX, does the change in http-spec impact this test?

File: examples/petstore.oas2.yaml (hunk 1 above)
Comment (XVincentX) [reply examples/petstore.oas2.yaml]: Argh, I must have done something wrong here. I'll restore it.

File: test-harness/specs/validate-body-params/no-consumes.defined.oas2.txt (hunk 2 above)
Comment (XVincentX) [reply test-harness/specs/validate-body-params/no-consumes.defined.oas2.txt]: Yeah this is something we need to discuss with @philsturgeon 

basically a spec file that has neither a global or a local consumes is wrong — and this test was testing a behaviour which should not happen.

If you do not define what your API can consume, we can't even try to parse and validate your body.

Comment (philsturgeon) [reply test-harness/specs/validate-body-params/no-consumes.defined.oas2.txt]: This seems like reasonable thinking, but if we want to be friendly we can say “they might have meant JSON” perhaps?

Comment (XVincentX) [reply test-harness/specs/validate-body-params/no-consumes.defined.oas2.txt]: Given this change will affect other parts of Stoplight I've put a comment in the relevant PR, let's continue there @philsturgeon https://github.com/stoplightio/http-spec/pull/25#issuecomment-535035897

**Pull request merged**